icarusalg/Geometry/CMakeLists.txt

```diff
@@ -1,17 +1,20 @@
 
-cet_make(
-  SUBDIRS "details"
+file(GLOB src_files *.cxx details/*.cxx)
+cet_make_library(
+  SOURCE ${src_files}
   LIBRARIES
-          larcorealg_Geometry
-          larcorealg_CoreUtils
-          ${MF_MESSAGELOGGER}
-          ${FHICLCPP}
-          cetlib
-          cetlib_except
+          larcorealg::Geometry
+          larcorealg::CoreUtils
+          larcoreobj::SimpleTypesAndConstants
+          messagefacility::MF_MessageLogger
+          fhiclcpp::fhiclcpp
+          cetlib::cetlib
+          cetlib_except::cetlib_except
           ROOT::Core
           ROOT::Physics
           ROOT::Geom
           ROOT::GenVector
+          CLHEP::CLHEP
```

Comment:
Dependencies should be divided into `PUBLIC` and `PRIVATE` depending on whether they are transitive (conferred by header use) or not (used only by the compilation unit).